src/ddmd/root/filename.d

```diff
@@ -721,3 +722,112 @@ nothrow:
         return str;
     }
 }
+
+version(Windows)
+{
+    /*
+      The code before used the POSIX function `mkdir` on Windows. That
+      function is now deprecated and fails with long paths, so instead
+      we use the newer `CreateDirectoryW`.
+
+      `CreateDirectoryW` is the unicode version of the generic macro
+      `CreateDirectory`.  `CreateDirectoryA` has a file path
+      limitation of 248 characters, `mkdir` fails with less and might
+      fail due to the number of consecutive `..`s in the
+      path. `CreateDirectoryW` also normally has a 248 character
+      limit, unless the path is absolute and starts with `\\?\`. Note
+      that this is different from starting with the almost identical
+      `\\?`.
+
+      Please consult
+      https://msdn.microsoft.com/en-us/library/windows/desktop/aa363855(v=vs.85).aspx
+    */
+    private int _mkdir(const(char)* path) nothrow
+    {
+        import core.sys.windows.winbase: CreateDirectoryW, GetLastError;
+        import core.sys.windows.winerror: ERROR_ALREADY_EXISTS;
+        import core.stdc.errno: errno, EEXIST;
+
+        const createRet = path.extendedPathThen!(p => CreateDirectoryW(&p[0],
+                                                                       null /*securityAttributes*/));
+        const lastError = GetLastError();
+
+        // Preserve compatibility with mkdir since the calling code expects
+        // errno to be set.
+        if (createRet == 0 && lastError == ERROR_ALREADY_EXISTS)
+            errno(EEXIST);
```

Comment:
but not on any other error? `ERROR_PATH_NOT_FOUND` -> `ENOENT`

Reply:
Fixed.